plasmapy/plasma/grids.py

```diff
@@ -989,54 +989,53 @@ def interpolate_indices(self, pos: Union[np.ndarray, u.Quantity]):
             `~numpy.ndarray` is provided, units will be assumed to match
             those of the grid.
 
-        Returns
-        -------
-        i : `~numpy.ndarray`, shape (n,3)
-            An array of indices corresponding to the positions such that
-            ``i[n,:] = ix,iy,iz`` such that ``grid[ix,iy,iz,:]`` ∼ ``pos[n,:]``
+        *args : `str`
+            Strings that correspond to DataArrays in the dataset
 
+        persistent : `bool`
+            If `True`, the interpolator will assume the grid and its
+            contents have not changed since the last interpolation. This
+            substantially speeds up the interpolation when many
+            interpolations are performed on the same grid in a loop.
+            ``persistent`` overrides to `False` if the arguments list
+            has changed since the last call.
         """
-        # Condition pos
-        # If a single point was given, add empty dimension
-        if pos.ndim == 1:
-            pos = np.reshape(pos, [1, 3])
-        pos2 = np.zeros(pos.shape)
-        # Convert position to SI and then strip units
-        if hasattr(pos, "unit"):
-            pos2 = pos.si.value
-        else:
-            for i in range(3):
-                pos2[:, i] = (pos[:, i] * self.units[i]).si.value
-
-        # Interpolate indices
-        i = self.interpolator(pos2)
-
-        # TODO: Check interpolated positions and reject any (set to NaN)
-        # that are above a certain tolerance distance?
-        # currently the nonuniform interpolator can't tell when a value
-        # is out of bounds...
-
-        # Note: i contains nan values which must be replaced with 0's with
-        # appropriate units in the second layer interpolator functions.
 
-        return i
+        raise NotImplementedError(
+            "Volume-averaged interpolator is not yet " "implemented for this grid type."
+        )
```

Comment:
If this is a special case in the CartesianGrid, do we want this in the base class, or do we just want to move it into that subclass? I guess you also expect other eventual subclasses to implement this one...?

Reply:
Yes, although probably not all of them (very computationally intensive on a non-uniform grid, I think). These algorithms have to be custom-written for each grid since they rely on the specific geometry.